### PR TITLE
[IMP] mail: User avatar in public page

### PR DESCRIPTION
**Pull request by maki-odoo**

### CODE REVIEW

File: addons/mail/static/src/new/discuss/discuss.js [hunk 1]
```diff
@@ -165,12 +165,16 @@ export class Discuss extends Component {
 
     async renameGuest({ value: name }) {
         const newName = name.trim();
-        if (this.store.guest?.name !== newName) {
+        if (this.store.guest && this.store.guest.name !== newName) {
```
Comment (zel-odoo) [inline addons/mail/static/src/new/discuss/discuss.js]: It is the same I think.

File: addons/mail/static/src/new/discuss/discuss.xml [hunk 2]
```diff
@@ -40,12 +40,12 @@
                     <button t-if="thread.type !== 'mailbox'" class="btn" t-ref="settings" title="Show Call Settings" t-on-click="toggleSettings">
                         <i class="fa fa-lg fa-gear text-700"/>
                     </button>
-                    <div t-if="store.guest and !env.isSmall" class="d-flex align-items-center">
-                        <img class="o-mail-discuss-guest-avatar mx-1 rounded-circle o_object_fit_cover flex-shrink-0" alt="Avatar" t-att-src="guestAvatarUrl"/>
+                    <div t-if="props.public and !env.isSmall" class="d-flex align-items-center">
+                        <img class="o-mail-discuss-guest-avatar mx-1 rounded-circle o_object_fit_cover flex-shrink-0" alt="Avatar" t-att-src="avatarUrl"/>
```
Comment (zel-odoo) [inline addons/mail/static/src/new/discuss/discuss.xml]: `o-mail-discuss-guest-avatar` is not a correct name for user

File: addons/mail/static/src/new/discuss/discuss.xml [hunk 3]
```diff
@@ -40,12 +40,12 @@
...
                         <AutoresizeInput
                             className="'o-mail-discuss-thread-guest-name lead fw-bold flex-shrink-1 text-dark'"
                             onValidate.bind="renameGuest"
-                            value="store.guest.name"
+                            value="store.guest?.name or store.user?.name"
```
Comment (zel-odoo) [inline addons/mail/static/src/new/discuss/discuss.xml]: functionally, we should only allow guests to change names here and not the user. (like what we did before)

Comment (alexkuhn) [inline addons/mail/static/src/new/discuss/discuss.xml:48]: Use `store.self` which is either one of them

File: addons/mail/static/src/new/discuss/discuss.js [hunk 4]
```diff
@@ -165,12 +165,16 @@ export class Discuss extends Component {
...
     }
 
-    get guestAvatarUrl() {
-        return `/mail/channel/${this.thread.id}/guest/${this.store.guest.id}/avatar_128?unique=${this.store.guest.name}`;
+    get avatarUrl() {
+        if (this.store.guest) {
+            return `/mail/channel/${this.thread.id}/guest/${this.store.guest.id}/avatar_128?unique=${this.store.guest.name}`;
+        } else {
+            return this.store.user.avatarUrl;
+        }
```
Comment (alexkuhn) [inline addons/mail/static/src/new/discuss/discuss.js:178]: This becomes almost the same as `avatarUrl` in `persona_model`.

It goes in `mail/channel` route for guest just for access rights.
Would be nice to clean code of `avatarUrl`, as there are too many different uses that should be reconciled

File: addons/mail/static/src/new/discuss/discuss.js (hunk 1 above)
Comment (alexkuhn) [reply addons/mail/static/src/new/discuss/discuss.js]: Not really, if `newName` undefined and no store guest, it passes

File: addons/mail/static/src/new/discuss/discuss.xml (hunk 3 above)
Comment (alexkuhn) [reply addons/mail/static/src/new/discuss/discuss.xml]: Agree, rename of user has more implication, it cannot be simply changed here. Better not allow renaming of user in public page

File: addons/mail/static/src/new/discuss/discuss.xml (hunk 2 above)
Comment (alexkuhn) [reply addons/mail/static/src/new/discuss/discuss.xml]: Indeed, it's just self avatar

File: addons/mail/static/src/new/discuss/discuss.js (hunk 1 above)
Comment (alexkuhn) [reply addons/mail/static/src/new/discuss/discuss.js]: But it assumes we allow renaming of user, which is not desirable here

**Review by alexkuhn — CHANGES_REQUESTED** [review]
Ok to show the avatar, but not ok to let user change name in public page. Only guests should be allowed to do so

File: addons/mail/static/src/new/discuss/discuss.js (hunk 4 above)
Comment (maki-odoo) [reply addons/mail/static/src/new/discuss/discuss.js:178]: Follow up PR :)

File: addons/mail/static/src/new/discuss/discuss.xml [hunk 5]
```diff
@@ -40,13 +40,18 @@
...
-                            onValidate.bind="renameGuest"
-                            value="store.guest.name"
-                        />
+                    <div t-if="props.public and !env.isSmall" class="d-flex align-items-center">
+                        <img class="o-mail-discuss-self-avatar mx-1 rounded-circle o_object_fit_cover flex-shrink-0" alt="Avatar" t-att-src="avatarUrl"/>
+                        <div class="lead fw-bold flex-shrink-1 text-dark">
+                            <t t-if="store.user" t-esc="store.user.name"/>
+                            <t t-else="">
+                                <AutoresizeInput
+                                    className="'o-mail-discuss-thread-guest-name'"
```
Comment (alexkuhn) [inline addons/mail/static/src/new/discuss/discuss.xml]: classname is actually unused, so I removed it

**Pull request merged**